lib/mock/mock-agent.js

```diff
@@ -0,0 +1,112 @@
+'use strict'
+
+const { kAgentOpts, kAgentCache } = require('../core/symbols')
+const { Agent } = require('../agent')
+const {
+  kAgent,
+  kMockAgentSet,
+  kMockAgentGet,
+  kDispatches,
+  kCreatePool,
+  kIsMockActive,
+  kNetConnect,
+  kGetNetConnect
+} = require('./mock-symbols')
+const MockClient = require('./mock-client')
+const MockPool = require('./mock-pool')
+const { matchValue } = require('./mock-utils')
+const { InvalidArgumentError } = require('../core/errors')
+
+class MockAgent {
+  constructor (opts) {
+    this[kNetConnect] = true
+    this[kIsMockActive] = true
+
+    // Instantiate Agent and encapsulate
+    const agent = new Agent(opts)
```

Comment:
This should be an option instead.

Reply:
Sure thing, I'll include this in the next update with typings and docs! :) Just to clarify one thing, do you mean a mandatory option (e.g. `opts.agent` and fail if it's not present) or an option with a default (e.g. `opts.agent || new Agent(opts)`)?

Reply:
I'm ok with an option with a default, that'd be a nicer user experience.

Reply:
Okay no probs! :)